src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -165,6 +165,11 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
       const current = finishedWork.alternate;
       commitWork(current, finishedWork);
     }
+    // if the root is a HostContainer, it may have a callback.
+    if (finishedWork.tag === HostContainer) {
+      const current = finishedWork.alternate;
+      commitLifeCycles(current, finishedWork);
```

Comment:
This needs to move in behind the effectTag check. Because we don't want to invoke this unless the root updates. This can happen if a render is scheduled, but we do a deep setState update at a higher priority first.

This isn't actually possible right now without #7457 but it will be.